lua/telescope/pickers/entry_display.lua

```diff
@@ -1,15 +1,27 @@
+local utils = require('telescope.utils')
+
 local entry_display = {}
 
 local function truncate(str, len)
   str = tostring(str) -- We need to make sure its an actually a string and not a number
-  -- TODO: This doesn't handle multi byte chars...
-  if vim.fn.strdisplaywidth(str) > len then
-    str = str:sub(1, len - 1)
-    str = str .. "…"
+  if utils.strdisplaywidth(str) <= len then
+    return str
+  end
+  local charlen = 0
+  local result = ''
+  while true do
+    local part = utils.strcharpart(str, 0, charlen) ..  '…'
+    if utils.strdisplaywidth(part) > len then
+      break
+    end
+    result = part
+    charlen = charlen + 1
```

Comment:
Hmmm isn't it not better to only look at the new part step by step and memorize it the prev result? I am just a little bit worried about performance :) I rewrote it to that: Any thoughts? Tests still pass
```suggestion
  local charlen = 0
  local cur_len = 0
  local result = ''
  local len_of_dots = utils.strdisplaywidth('…')
  while true do
    local part = utils.strcharpart(str, charlen, 1)
    cur_len = cur_len + utils.strdisplaywidth(part)
    if (cur_len + len_of_dots) > len then
      result = result .. '…'
      break
    end
    result = result .. part
    charlen = charlen + 1
  end
```

Reply:
Look good! committed on 1f77e33